cmd/lncli/commands.go

```diff
@@ -2286,9 +2295,7 @@ func sendPayment(ctx *cli.Context) error {
 			return err
 		}
 
-		req.DestCustomRecords = map[uint64][]byte{
-			record.KeySendType: preimage[:],
-		}
+		req.DestCustomRecords[record.KeySendType] = preimage[:]
```

Comment:
What should we do if this overwrites a user supplied preimage? It may be confusing if users supplies one preimage and it isn’t used.

We could change to only generating one if this field isn’t set or simply fail.

The bigger question in my mind is if we want to allow user supplied preimages altogether. The former may resolve some complications in tracking keysend payments, perhaps it is okay to permit it given we already allow arbitrary preimages for hodl invoices. Thoughts?

Reply:
I think it is okay for this experimental release of key send. Not only was it already possible to choose a preimage for hodl invoices, but one can even do this for normal invoices via `lncli addinvoice --preimage`. 

When we unify key send with moon amp, it seems like a good time to me to rethink where the preimage is generated.

Reply:
The current implementation lets a user specified preimage override the generated one. Added comment to clarify that.